### DISCUSSION TITLE
checkoutPaymentCreate results in "You do not have permission to perform this action"

### DISCUSSION THREAD

**Discussion by rolego** [saleor/saleor#9379] [Q&A] [answered]
Dear all

I'm trying to integrate stripe as payment gateway. I'm currently stuck at the following situation:
The mutation checkoutPaymentCreate(..) results in an error with message "You do not have permission to perform this action" but no further information.

Does anyone have an idea how to track down problems like this?

Thank you for any hint.

**Comment by rolego** [comment, marked as answer]
I was meanwhile able to solve the problem by removing  payment/metadata  from the GQL subselection. I think it's a bit unfortunate that an issue like this can consume that much  effort to solve.

**Comment by maarcingebala** [comment]
Hey, I think it should give you the information about which field triggers the error like so:
```
"message": "You do not have permission to perform this action",
      "locations": [
        {
          "line": 8,
          "column": 7
        }
      ],
      "path": [
        "checkoutPaymentCreate",
        "payment",
        "metadata"
      ]
```

**Reply by rolego** [reply]
Indeed! And it is even documented: https://docs.saleor.io/docs/3.x/developer/api-conventions/error-handling#permission-errors

Thank you Marcin and sorry for this. 🤦
